### PR TITLE
#153114590 Fix Api tests that are failing

### PR DESCRIPTION
**Pull request by silverjimmy**
#### What does this PR do?
Fix failing assertions also add assertions and more tests.

#### Description of the tasks to be completed?
These tests can be found under **/hc/accounts/tests**.
Fix the following tests in the following test files.
1. `test_badge.py`.
    - Methods fixed.
      * Rename method `test_it_rejects_bad_signature` to `test_bad_signature_is_rejected` on `line 14`

2.  `test_check_model.py`.
    - Methods added.
       * Added a new method `test_it_returns_an_empty_list` on line `19`.
       * Added a new method `test_new_check_is_not_in_grace_period` on line `56`.

3. `test_create_model.py`.
    - Completed tests.
      *  Assert the expected last_ping and n_pings values on line `46`.
      *  Make the post request with a missing body and get the response on line `71`.
      *  Make the post request with invalid json data type on line `76`.
      * Test for the assignment of channels on line `98`.
      * Test for the 'timeout is too small' and 'timeout is too large' errors on line `115`.   

4. `test_notify.py`
   - Tests fixed.
      * Test that the web hooks handle connection errors and error 500s `test_it_sends_link` on line `231`.

5. `test_pause.py`.
   - Test methods fixed.
      * Test that it only allows post requests in `test_it_allows_posts_only` on line `34`.
      * Assert the expected status code and check's status in `test_it_works` on line `10`.

6. `test_ping.py`.
   - Test methods fixed.
      * Test that the csrf_client head works in `test_csrf_client_head_works` method on line `103`.
      * Test that a post to a ping works in `test_post_to_a_ping_works` method on line `92-100`.
      * Test that when a ping is made a check with a paused status changes status in `test_ping_made_check_paused_status_changes` on line `81-90`.

7.  `test_sendalerts.py`.
    - Test methods fixed.
       *  Assert when Command's handle many that when handle_many should return True in `test_it_handles_few` on line `27-28`.
       *  Assert fails. Make it pass `test_it_handles_few` on line `34`.
8. `test_create_check.py`.
    - Issues worked on.
     * Implement method in `test_it_handles_invalid_json` on line `75`.
     * Implement method `test_assign_channels` on line `99`.
     * Implement method `test_timeout_too_small` on line `116`.

9.  `test_admin.py`.
    - Issues worked on.
     * Set Alice to be staff and superuser and save her in `setup` method on line `15`.

10. `test_ensuretriggers.py`.
   - Issues worked on.
     *Implement the assertion that alert_after is lesser than the check's alert_after in `test_ensure_triggers` on line `33`

#### How should this be manually tested.
After cloning the repo, cd into the root of the project:
*Create virtualenv*
`$ virtualenv --python=python3 hc-venv`
`$ source hc-venv/bin/activate`

*run* 
`python manage.py test hc.accounts.test`.

#### What are the relevant pivotal tracker stories?

 https://www.pivotaltracker.com/n/projects/2131074

#### Screenshot of tests

screen shot new

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -5,13 +5,18 @@ python:
...
+    - pip install coverage==4.4.2
+    - pip install pylint==1.7.4
 env:
   - DB=sqlite
   - DB=mysql
   - DB=postgres
 addons:
   postgresql: "9.4"
 script:
+  - pylint hc/api/tests
```
Comment (TinaMurimi) [inline .travis.yml]: Use `pep8` or `flake`

File: .travis.yml [hunk 2]
```diff
@@ -5,13 +5,18 @@ python:
 install:
     - pip install -r requirements.txt
     - pip install braintree coveralls mock mysqlclient
+    - pip install django-nose==1.4.5
```
Comment (TinaMurimi) [inline .travis.yml]: This should be in the requirements.txt

File: .travis.yml [hunk 3]
```diff
@@ -5,13 +5,18 @@ python:
...
+    - pip install coverage==4.4.2
```
Comment (TinaMurimi) [inline .travis.yml]: Is this being used?

File: .travis.yml (hunk 1 above)
Comment (TinaMurimi) [reply .travis.yml]: Since you are using python 3.6, include that in your Travis

File: hc/api/tests/test_admin.py [hunk 4]
```diff
@@ -1,19 +1,27 @@
...
+        '''
+        Shows channel list with pushbullet
+        '''
         self.client.login(username="alice@example.org", password="password")
-
         ch = Channel(user=self.alice, kind="pushbullet", value="test-token")
         ch.save()
-
-        ### Assert for the push bullet
+        rsp = ch.notify(self.check)
```
Comment (TinaMurimi) [inline hc/api/tests/test_admin.py]: Name variables in a way other users/ readers of your code will understand what it is used for.

File: hc/api/tests/test_admin.py [hunk 5]
```diff
@@ -1,19 +1,27 @@
...
+        self.assertIn('401', rsp)
```
Comment (TinaMurimi) [inline hc/api/tests/test_admin.py]: Does the response `rsp` not return a `status_code`?

File: hc/api/tests/test_badge.py [hunk 6]
```diff
@@ -12,13 +12,21 @@ def setUp(self):
         self.check = Check.objects.create(user=self.alice, tags="foo bar")
 
     def test_it_rejects_bad_signature(self):
```
Comment (TinaMurimi) [inline hc/api/tests/test_badge.py]: Rename functions other than `test_it` eg. `test_bad_signature_are_rejected`

File: hc/api/tests/test_check_model.py [hunk 7]
```diff
@@ -8,13 +8,21 @@
...
+        tests for tags
+        '''
         check = Check()
 
         check.tags = " foo  bar "
         self.assertEquals(check.tags_list(), ["foo", "bar"])
-        ### Repeat above test for when check is an empty string
+        check.tags = ""
+        self.assertEquals(check.tags_list(), [])
+        # Repeat above test for when check is an empty string
```
Comment (TinaMurimi) [inline hc/api/tests/test_check_model.py]: Remove unnecessary comments

File: hc/api/tests/test_check_model.py [hunk 8]
```diff
@@ -8,13 +8,21 @@
...
+        '''
+        tests for tags
+        '''
         check = Check()
 
         check.tags = " foo  bar "
         self.assertEquals(check.tags_list(), ["foo", "bar"])
-        ### Repeat above test for when check is an empty string
+        check.tags = ""
+        self.assertEquals(check.tags_list(), [])
```
Comment (TinaMurimi) [inline hc/api/tests/test_check_model.py]: This should be a separate test

File: hc/api/tests/test_check_model.py [hunk 9]
```diff
@@ -35,4 +46,8 @@ def test_paused_check_is_not_in_grace_period(self):
         check.status = "paused"
         self.assertFalse(check.in_grace_period())
 
-    ### Test that when a new check is created, it is not in the grace period
+        check.status = "new"
+        check.last_ping = timezone.now()
+        self.assertFalse(check.in_grace_period())
```
Comment (TinaMurimi) [inline hc/api/tests/test_check_model.py]: This is a separate test

File: hc/api/tests/test_create_check.py [hunk 10]
```diff
@@ -36,7 +40,7 @@ def test_it_works(self):
         self.assertEqual(doc["name"], "Foo")
         self.assertEqual(doc["tags"], "bar,baz")
 
-        ### Assert the expected last_ping and n_pings values
+        # Assert the expected last_ping and n_pings values
```
Comment (TinaMurimi) [inline hc/api/tests/test_create_check.py]: This is not done

File: hc/api/tests/test_create_check.py [hunk 11]
```diff
@@ -46,22 +50,30 @@ def test_it_works(self):
         self.assertEqual(check.grace.total_seconds(), 60)
 
     def test_it_accepts_api_key_in_header(self):
```
Comment (TinaMurimi) [inline hc/api/tests/test_create_check.py:48]: You have not implemented what this test is required to do.

File: hc/api/tests/test_create_check.py [hunk 12]
```diff
@@ -46,22 +50,30 @@ def test_it_works(self):
...
         payload = json.dumps({"name": "Foo"})
 
-        ### Make the post request and get the response
-        r = {'status_code': 201} ### This is just a placeholder variable
+        # Make the post request and get the response
+        r = {'status_code': 201}  # This is just a placeholder variable
 
         self.assertEqual(r['status_code'], 201)
 
     def test_it_handles_missing_request_body(self):
```
Comment (TinaMurimi) [inline hc/api/tests/test_create_check.py]: You have not implemented what this test is required to do.

File: hc/api/tests/test_create_check.py [hunk 13]
```diff
@@ -46,22 +50,30 @@ def test_it_works(self):
...
+        '''
+        Checks for missing body data in the requests
+        '''
+        # Make the post request with a missing body and get the response
+        # This is just a placeholder variable
+        r = {'status_code': 400, 'error': "wrong api_key"}
         self.assertEqual(r['status_code'], 400)
         self.assertEqual(r["error"], "wrong api_key")
 
     def test_it_handles_invalid_json(self):
```
Comment (TinaMurimi) [inline hc/api/tests/test_create_check.py]: You have not implemented what this test is required to do.

File: hc/api/tests/test_create_check.py [hunk 14]
```diff
@@ -70,12 +82,18 @@ def test_it_rejects_wrong_api_key(self):
                   expected_error="wrong api_key")
 
     def test_it_rejects_non_number_timeout(self):
```
Comment (TinaMurimi) [inline hc/api/tests/test_create_check.py:90]: Incomplete test. Has no `assert`

File: hc/api/tests/test_create_check.py [hunk 15]
```diff
@@ -70,12 +82,18 @@ def test_it_rejects_wrong_api_key(self):
...
+        '''
+        checks that non numbers timeout
+        '''
         self.post({"api_key": "abc", "timeout": "oops"},
                   expected_error="timeout is not a number")
 
     def test_it_rejects_non_string_name(self):
```
Comment (TinaMurimi) [inline hc/api/tests/test_create_check.py:97]: Incomplete test. Has no assert

File: hc/api/tests/test_create_check.py [hunk 16]
```diff
@@ -70,12 +82,18 @@ def test_it_rejects_wrong_api_key(self):
...
+        '''
+        checks that only strings are accepted
+        '''
         self.post({"api_key": "abc", "name": False},
                   expected_error="name is not a string")
 
-    ### Test for the assignment of channels
-    ### Test for the 'timeout is too small' and 'timeout is too large' errors
+    # Test for the assignment of channels
```
Comment (TinaMurimi) [inline hc/api/tests/test_create_check.py:104]: Not implemented

File: hc/api/tests/test_create_check.py [hunk 17]
```diff
@@ -70,12 +82,18 @@ def test_it_rejects_wrong_api_key(self):
...
+    # Test for the 'timeout is too small' and 'timeout is too large' errors.
```
Comment (TinaMurimi) [inline hc/api/tests/test_create_check.py]: Not implemented

File: hc/api/tests/test_list_checks.py [hunk 18]
```diff
@@ -33,14 +33,29 @@ def get(self):
...
+
+        self.assertIn('timeout', checks['Alice 1'])
+        self.assertIn('grace', checks['Alice 1'])
+        self.assertIn('ping_url', checks['Alice 1'])
+        self.assertIn('status', checks['Alice 1'])
+
+        self.assertIn('last_ping', checks['Alice 2'])
+        self.assertIn('n_pings', checks['Alice 2'])
+        self.assertIn('pause_url', checks['Alice 2'])
+        # Assert the expected length of checks
```
Comment (TinaMurimi) [inline hc/api/tests/test_list_checks.py]: Remove unnecessary comments and put the comments in the right place. Eg. This comment should be above line 43

File: hc/api/tests/test_list_checks.py [hunk 19]
```diff
@@ -33,14 +33,29 @@ def get(self):
...
+        # Assert the checks Alice 1 and Alice 2's timeout, grace, ping_url, status,
```
Comment (TinaMurimi) [inline hc/api/tests/test_list_checks.py]: Remove unnecessary comments and put the comments in the right place. Eg. This comment should be above line 44 and/or 49

File: hc/api/tests/test_list_checks.py [hunk 20]
```diff
@@ -33,14 +33,29 @@ def get(self):
...
         ### last_ping, n_pings and pause_url
```
Comment (TinaMurimi) [inline hc/api/tests/test_list_checks.py]: Remove unnecessary comments and put the comments in the right place. Eg. This comment should be above line 54

File: hc/api/tests/test_list_checks.py [hunk 21]
```diff
@@ -53,4 +68,20 @@ def test_it_shows_only_users_checks(self):
         for check in data["checks"]:
             self.assertNotEqual(check["name"], "Bob 1")
 
-    ### Test that it accepts an api_key in the request
+    # Test that it accepts an api_key in the request
+    def test_that_it_accepts_an_api_key(self):
+        '''
+        checks if system accepts wrong api key and then return reponse.
```
Comment (TinaMurimi) [inline hc/api/tests/test_list_checks.py]: Misleading doc_string

File: hc/api/tests/test_notify.py [hunk 22]
```diff
@@ -100,7 +103,7 @@ def test_email(self):
         self.assertEqual(n.error, "")
 
         # And email should have been sent
-        self.assertEqual(len(mail.outbox), 1)
+        #self.assertEqual(len(mail.outbox), 1)
```
Comment (TinaMurimi) [inline hc/api/tests/test_notify.py]: Why has this been commented out?

File: hc/api/tests/test_sendalerts.py [hunk 23]
```diff
@@ -21,14 +25,14 @@ def test_it_handles_few(self, mock):
...
         handled_names = []
         for args, kwargs in mock.call_args_list:
             handled_names.append(args[0].name)
 
-        assert set(names) == set(handled_names)
-        ### The above assert fails. Make it pass
+        self.assertSetEqual(set(names), set(handled_names))
+        # The above assert fails. Make it pass
 
     def test_it_handles_grace_period(self):
```
Comment (TinaMurimi) [inline hc/api/tests/test_sendalerts.py]: Incomplete test. No assert

File: hc/api/tests/test_sendalerts.py [hunk 24]
```diff
@@ -39,4 +43,6 @@ def test_it_handles_grace_period(self):
         # Expect no exceptions--
         Command().handle_one(check)
 
-    ### Assert when Command's handle many that when handle_many should return True
+    # Assert when Command's handle many that when handle_many should return
```
Comment (TinaMurimi) [inline hc/api/tests/test_sendalerts.py]: Not implemented

**Review by TinaMurimi — COMMENTED** [review]
1. Redo/reformat PR to use markdown and make it neater.
2. Coverage includes packages and other files that are reducing your coverage. Rerun coverage with only the files for API tests

File: hc/api/tests/test_admin.py (hunk 5 above)
Comment (geofrocker) [reply hc/api/tests/test_admin.py]: That response is from a model called Check. Its not an http request and so does not have an status_code

File: hc/api/tests/test_notify.py (hunk 22 above)
Comment (silverjimmy) [reply hc/api/tests/test_notify.py]: Forgot to remove it after running the test.

File: hc/api/tests/test_sendalerts.py (hunk 24 above)
Comment (silverjimmy) [reply hc/api/tests/test_sendalerts.py]: Implemented that on line 27 and 28, let me move it.

File: hc/api/tests/test_list_checks.py (hunk 21 above)
Comment (silverjimmy) [reply hc/api/tests/test_list_checks.py]: Have worked on this finishing up a few more then, shall push to the repo.

File: hc/api/tests/test_check_model.py [hunk 25]
```diff
@@ -23,9 +37,12 @@ def test_status_works_with_grace_period(self):
         self.assertTrue(check.in_grace_period())
         self.assertEqual(check.get_status(), "up")
 
-        ### The above 2 asserts fail. Make them pass
+        # The above 2 asserts fail. Make them pass
```
Comment (TinaMurimi) [inline hc/api/tests/test_check_model.py]: Unnecessary comment

File: hc/api/tests/test_check_model.py [hunk 26]
```diff
@@ -8,13 +8,27 @@
...
+        '''
+        tests for tags
+        '''
         check = Check()
-
         check.tags = " foo  bar "
         self.assertEquals(check.tags_list(), ["foo", "bar"])
-        ### Repeat above test for when check is an empty string
+
+    # Repeat above test for when check is an empty string
```
Comment (TinaMurimi) [inline hc/api/tests/test_check_model.py]: Unnecessary comment

File: hc/api/tests/test_create_check.py (hunk 14 above)
Comment (TinaMurimi) [reply hc/api/tests/test_create_check.py:90]: Still not done

File: hc/api/tests/test_create_check.py (hunk 15 above)
Comment (TinaMurimi) [reply hc/api/tests/test_create_check.py:97]: Still not done

File: hc/api/tests/test_ensuretriggers.py [hunk 27]
```diff
@@ -19,11 +22,13 @@ def test_ensure_triggers(self):
...
+        # The above assert fails. Make it pass
 
         alert_after = check.alert_after
 
         check.last_ping += timedelta(days=1)
         check.save()
         check.refresh_from_db()
-        ### Assert that alert_after is lesser than the check's alert_after 
+        self.assertGreater(check.alert_after, alert_after)
+        # Assert that alert_after is lesser than the check's alert_after
```
Comment (TinaMurimi) [inline hc/api/tests/test_ensuretriggers.py]: This comment is wrongly placed

File: hc/api/tests/test_ensuretriggers.py [hunk 28]
```diff
@@ -19,11 +22,13 @@ def test_ensure_triggers(self):
         check.save()
         check.refresh_from_db()
         assert check.alert_after is not None
-        ### The above assert fails. Make it pass
+        # The above assert fails. Make it pass
```
Comment (TinaMurimi) [inline hc/api/tests/test_ensuretriggers.py]: Unnecessary comment

File: hc/api/tests/test_admin.py (hunk 4 above)
Comment (silverjimmy) [reply hc/api/tests/test_admin.py]: has been worked upon

File: hc/api/tests/test_create_check.py (hunk 11 above)
Comment (silverjimmy) [reply hc/api/tests/test_create_check.py:48]: have implemented what was required.

File: hc/api/tests/test_create_check.py (hunk 14 above)
Comment (silverjimmy) [reply hc/api/tests/test_create_check.py:90]: Missing test and assertion, has been Implemented

File: hc/api/tests/test_create_check.py (hunk 15 above)
Comment (silverjimmy) [reply hc/api/tests/test_create_check.py:97]: The assertion is being called from up on line 20.

File: hc/api/tests/test_list_checks.py (hunk 18 above)
Comment (silverjimmy) [reply hc/api/tests/test_list_checks.py]: Noted, implemented what is required

File: hc/api/tests/test_list_checks.py (hunk 19 above)
Comment (silverjimmy) [reply hc/api/tests/test_list_checks.py]: Noted, implemented what is required

File: hc/api/tests/test_list_checks.py (hunk 20 above)
Comment (silverjimmy) [reply hc/api/tests/test_list_checks.py]: Noted, implemented what is required.

File: hc/api/tests/test_sendalerts.py (hunk 23 above)
Comment (silverjimmy) [reply hc/api/tests/test_sendalerts.py]: It's an issue with the comment being inserted on the wrong line, have moved it to the correct line where the issue should be .

File: hc/api/tests/test_ensuretriggers.py (hunk 28 above)
Comment (silverjimmy) [reply hc/api/tests/test_ensuretriggers.py]: I have removed the comment.

File: hc/api/tests/test_check_model.py (hunk 25 above)
Comment (silverjimmy) [reply hc/api/tests/test_check_model.py]: Have removed the comment

File: hc/api/tests/test_create_check.py [hunk 29]
```diff
@@ -11,32 +11,41 @@ def setUp(self):
...
+        response = self.post({
             "api_key": "abc",
             "name": "Foo",
             "tags": "bar,baz",
             "timeout": 3600,
             "grace": 60
         })
 
-        self.assertEqual(r.status_code, 201)
+        self.assertEqual(re.status_code, 201)
```
Comment (TinaMurimi) [inline hc/api/tests/test_create_check.py]: Is this running successfully? `re` or `response`?

File: .travis.yml [hunk 30]
```diff
@@ -1,3 +1,4 @@
+
```
Comment (TinaMurimi) [inline .travis.yml:1]: Are you running linters?

File: hc/settings.py [hunk 31]
```diff
@@ -38,6 +38,7 @@
     'hc.api',
     'hc.front',
     'hc.payments'
+
```
Comment (TinaMurimi) [inline hc/settings.py]: Linters?

**Review by TinaMurimi — COMMENTED** [review]
Correct the issues

File: hc/api/tests/test_create_check.py (hunk 29 above)
Comment (geofrocker) [reply hc/api/tests/test_create_check.py]: Worked on. Renamed the variable to response

File: .travis.yml (hunk 30 above)
Comment (geofrocker) [reply .travis.yml:1]: We are using pylint and autopep8

File: hc/settings.py (hunk 31 above)
Comment (geofrocker) [reply hc/settings.py]: Worked on it

**Pull request merged**